numpy/distutils/ccompiler.py

```diff
@@ -136,20 +137,39 @@ def CCompiler_spawn(self, cmd, display=None):
         if is_sequence(display):
             display = ' '.join(list(display))
     log.info(display)
-    s, o = exec_command(cmd)
-    if s:
-        if is_sequence(cmd):
-            cmd = ' '.join(list(cmd))
-        try:
-            print(o)
-        except UnicodeError:
-            # When installing through pip, `o` can contain non-ascii chars
-            pass
-        if re.search('Too many open files', o):
-            msg = '\nTry rerunning setup command until build succeeds.'
-        else:
-            msg = ''
-        raise DistutilsExecError('Command "%s" failed with exit status %d%s' % (cmd, s, msg))
+    try:
+        o = subprocess.check_output(cmd)
```

Comment:
`cmd` is documented to be a string, but in practice tends to be a list it seems; I originally tried to split on whitespace just in case, which Eric rightly asked me to remove -- I wonder if we'd consider adding a conditional  `shlex.split()` here for robustness given other recent similar changes.

Reply:
Also, the LGTM analysis wants me to remove assignment to `o` here, since it is never used.